### PR TITLE
vmware_guest - Add graceful guest shutdown/reboot support

### PR DESCRIPTION
**Pull request by Abdul-Anshad-A**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
vmware_guest

##### ANSIBLE VERSION
devel

##### SUMMARY
Add support for graceful guest shutdown/reboot operations.
fixes issue #3491

### CODE REVIEW

**Comment by Abdul-Anshad-A** [comment]
Migrated from https://github.com/ansible/ansible-modules-extras/pull/3507

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 1]
```diff
@@ -551,6 +551,19 @@ def set_powerstate(self, vm, state, force):
...
                                   'msg': "Cannot restart VM in the current state %s" % current_state}
+                        
+                elif expected_state in ['shutdownguest', 'rebootguest']:
+                    if current_state == 'poweredon' and vm.guest.toolsRunningStatus == 'guestToolsRunning':
+                        if expected_state == 'shutdownguest':
+                            task = vm.ShutdownGuest()
+                            result = {'changed': True, 'failed': False}
+                        else:
+                            task = vm.RebootGuest()
+                            result = {'changed': True, 'failed': False}
```
Comment (nerzhul) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py]: please verify task result to ensure it was changed or failed

**Review by nerzhul — CHANGES_REQUESTED** [review]

**Comment by Abdul-Anshad-A** [comment]
@nerzhul Corrected. I have no idea for the failing check.
Weird ! Triggered checks again & it passed.

**Comment by Abdul-Anshad-A** [comment]
ready_for_review

**Comment by nerzhul** [comment]
shipit

**Comment by dav1x** [comment]
shipit

**Comment by jctanner** [comment]
bot_status

**Comment by Abdul-Anshad-A** [comment]
Is there anything else required from my end to merge this change?

**Comment by jctanner** [comment]
@Abdul-Anshad-A you have a merge commit https://github.com/ansible/ansible/pull/20564/commits/d67c53f7f5e6250f0726fe7deb4d61dff2de3a69

Please rebase your branch: http://docs.ansible.com/ansible/dev_guide/developing_rebasing.html

@nerzhul also needs to approve or dismiss his change request.

**Comment by Abdul-Anshad-A** [comment]
I've re-based my branch.

**Comment by nerzhul** [comment]
needs_rebase

**Comment by Abdul-Anshad-A** [comment]
Re-based my branch again.

**Review by nerzhul — APPROVED** [review]
shipit

**Comment by jctanner** [comment]
bot_status

**Comment by dav1x** [comment]
shipit

**Comment by jctanner** [comment]
shipit

**Comment by jctanner** [comment]
@Abdul-Anshad-A thank you!

**Pull request merged**